File: ios/Classes/Serializer/NFCOptionsSerializer.swift
import Onfido

func getNFCConfiguration(_ nfcOption: String) -> NFCConfiguration {
    let nfcConfigurationMap: [String: NFCConfiguration] = [
        "DISABLED": .off,
        "OPTIONAL": .optional,
        "REQUIRED": .required
    ]
    return nfcConfigurationMap[nfcOption, default: .optional]
}

File: android/src/main/kotlin/com/onfido/sdk/flutter/serializer/NFCOptionsSerializer.kt
package com.onfido.sdk.flutter.serializer

import com.onfido.android.sdk.capture.model.NFCOptions
import com.onfido.android.sdk.capture.model.NFCOptions.Disabled
import com.onfido.android.sdk.capture.model.NFCOptions.Enabled.Optional
import com.onfido.android.sdk.capture.model.NFCOptions.Enabled.Required

internal fun getNFCOption(nfcOption: String): NFCOptions {
    val nfcOptionsMap = mapOf(
        "DISABLED" to Disabled,
        "OPTIONAL" to Optional,
        "REQUIRED" to Required
    )
    return nfcOptionsMap.getOrDefault(nfcOption, Optional)
}
